Assets/HoloToolkit/Input/Scripts/Utilities/ControllerFinder.cs

```diff
@@ -0,0 +1,99 @@
+
+    #if UNITY_WSA && UNITY_2017_2_OR_NEWER
+        public virtual InteractionSourceHandedness Handedness { get { return handedness; } set { handedness = value; } }
+        protected InteractionSourceHandedness handedness = InteractionSourceHandedness.Left;
+    
+    #endif
+
+        #region private members
+        protected MotionControllerInfo controller;
+        protected MotionControllerInfo.ControllerElementEnum element = MotionControllerInfo.ControllerElementEnum.PointingPose;
+        protected Transform elementTransform;
+        private bool isAttached = false;
+        #endregion
+
+
+        public virtual void OnEnable()
+        {
+    #if UNITY_WSA && UNITY_2017_2_OR_NEWER
+            // Look if the controller has loaded.
+            if (MotionControllerVisualizer.Instance.TryGetControllerModel(handedness, out controller))
+            {
+                AddControllerTransform(controller);
+            }
+
+            MotionControllerVisualizer.Instance.OnControllerModelLoaded += AddControllerTransform;
+            MotionControllerVisualizer.Instance.OnControllerModelUnloaded += RemoveControllerTransform;
+        }
+
+        protected virtual void OnDisable()
+        {
+            if (MotionControllerVisualizer.IsInitialized)
+            {
+                MotionControllerVisualizer.Instance.OnControllerModelLoaded -= AddControllerTransform;
+                MotionControllerVisualizer.Instance.OnControllerModelUnloaded -= RemoveControllerTransform;
+            }
+        }
+
+        protected virtual void OnDestroy()
+        {
+            if (MotionControllerVisualizer.IsInitialized)
+            {
+                MotionControllerVisualizer.Instance.OnControllerModelLoaded -= AddControllerTransform;
+                MotionControllerVisualizer.Instance.OnControllerModelUnloaded -= RemoveControllerTransform;
+            }
+        }
+
+        private void AddControllerTransform(MotionControllerInfo newController)
+        {
+            if (!isAttached && newController.Handedness == handedness)
+            {
+                if (!newController.TryGetElement(element, out elementTransform))
+                {
+                    Debug.LogError("Unable to find element of type " + element + " under controller " + newController.ControllerParent.name + "; not attaching.");
+                    return;
+                }
+                controller = newController;
+                // update elementTransform for cosnsumption
+                controller.TryGetElement(element, out elementTransform);
+                ElementTransform = elementTransform;
+
+                isAttached = true;
+            }
+        }
+
+        private void RemoveControllerTransform(MotionControllerInfo oldController)
+        {
+            if (isAttached && oldController.Handedness == handedness)
+            {
+                controller = null;
+                isAttached = false;
+            }
+
+        }
+    #endif
```

Comment:
This `#endif` causes there to be one too few braces in build targets other than UWP. It can be removed if my previous comment is implemented.